### PR TITLE
feat(email-otp): add change email flow with OTP

### PR DESCRIPTION
**Pull request by cdy-peters**
Implemented email change flow using email OTP's, this ensure consistent use of OTP's throughout the application.

The `overrideDefaultEmailVerification` option does not work with `changeEmail()` as `sendVerificationOTP()` requires a user with the provided email exists, as such, sending an OTP to the new email address is unsuccessful. 
Instead of implementing a workaround to address this, I believe implementing dedicated endpoints for the change email flow provides a  better DX as a developer may not want to override ***all*** email verification with OTP's.

I have added 2 endpoints, `emailOTP.requestEmailChangeOTP()` and `emailOTP.changeEmailOTP()`.
I have also added an option to enable verification of the current email before sending an OTP to the new email. This requires a valid OTP for the current email be passed to `emailOTP.requestEmailChangeOTP()`.

Closes #5540 #879 

---
## Summary by cubic
Adds a dedicated change-email-with-OTP flow with two authenticated endpoints. Supports optional current-email verification, triggers standard email verification callbacks, and refreshes the session on success.

- **New Features**
  - Endpoints: POST /email-otp/request-email-change and POST /email-otp/change-email (require a valid session).
  - Plugin option: changeEmail { enabled, verifyCurrentEmail }.
  - Rejects sendVerificationOtp with type "change-email"; use the new endpoints.
  - Added rate limits for both change-email endpoints; OTP types now include "change-email".
  - On success: updates the session cookie and sets emailVerified = true.
  - Docs and tests cover success, invalid/expired OTP, unauthorized/disabled flow, and edge cases.

- **Migration**
  - If you used sendVerificationOtp with type "change-email", switch to POST /email-otp/request-email-change and POST /email-otp/change-email, and set changeEmail.enabled = true.
  - To require confirming the current email first, set changeEmail.verifyCurrentEmail = true and verify with an email-verification OTP before requesting the change.

<sup>Written for commit d2f52082bfd171146fcfdb24af9c83ccb4314719. Summary will update on new commits.</sup>

### CODE REVIEW

File: docs/content/docs/plugins/email-otp.mdx [hunk 1]
```diff
@@ -241,6 +241,102 @@ type resetPasswordEmailOTP = {
...
+    ]
+})
+```
+
+By default, when a user requests to change their email, an OTP is sent to the **new** email address.
+The email is only updated after the user verifies the new email.
+
+#### Usage
+
+To change the user's email address with OTP, use the `emailOtp.requestEmailChange()` method to send a "change-email" OTP to the user's new email address.
```
[the remark this answers is not in the document]
Comment (cdy-peters) [reply docs/content/docs/plugins/email-otp.mdx:267]: `sendVerificationOtp` rejects `change-email` type as it is an invalid type due to a different format for the OTP identifier

File: packages/better-auth/src/plugins/email-otp/routes.ts [hunk 2]
```diff
@@ -1061,5 +1078,329 @@ export const resetPasswordEmailOTP = (opts: RequiredEmailOTPOptions) =>
...
+				throw APIError.from("BAD_REQUEST", ERROR_CODES.INVALID_OTP);
+			}
+			if (verificationValue.expiresAt < new Date()) {
+				await ctx.context.internalAdapter.deleteVerificationValue(
+					verificationValue.id,
+				);
+				throw APIError.from("BAD_REQUEST", ERROR_CODES.OTP_EXPIRED);
+			}
+
+			const [otpValue, attempts] = splitAtLastColon(verificationValue.value);
```
[the remark this answers is not in the document]
Comment (cdy-peters) [reply packages/better-auth/src/plugins/email-otp/routes.ts:1275]: Refactor in future pull request
